### PR TITLE
implemeted topological sort

### PR DESCRIPTION
**Pull request by yashsehgal29**
The code defines a C++ program to perform a topological sort on a directed graph. It uses an adjacency list representation with an unordered_map. The Graph class has methods to add edges, print the graph, and perform a topological sort. The topological sort is implemented using Depth-First Search (DFS). It starts by initializing an array to track visited nodes, then iterates through each node, invoking the DFS function to explore the graph, pushing nodes onto a stack as they are fully explored. Finally, it retrieves nodes from the stack to obtain the topologically sorted order of the graph. The program reads the number of nodes, edges, and edge information from the user and then displays the sorted order.

### CODE REVIEW

**Comment by CodingWarrior33** [comment]
Hi @yashsehgal29 , thanks a lot for taking time to contribute!!!
 
- Please add your Name and Roll No. in the comments
- Please add comments to explain what your code does

**Comment by yashsehgal29** [comment]
changes implemented as per the comments

**Comment by metabiswadeep** [comment]
@yashsehgal29 Please check spellings in the comments in the code.

**Comment by yashsehgal29** [comment]
done

File: Computational_Algorithms/Graphs/topological_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+     
+        for(auto i :map[node]){
+            if(!visited[i]){
+             dfs(i,visited,result);
+            }
+        }
+           result.push(node);
+    }
+    public:
+    //the topologocal sort
```
Comment (metabiswadeep) [inline Computational_Algorithms/Graphs/topological_sort.cpp:44]: wrong spelling

File: Computational_Algorithms/Graphs/topological_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#include<bits/stdc++.h>
+using namespace std;
+//B22147 Yash Sehgal
+//class for implementaion of Graphs
+class Graph{
+    unordered_map<int,list<int>> map;
+    int num_nodes;
+
+    public:
+    // s number of nodes
```
Comment (metabiswadeep) [inline Computational_Algorithms/Graphs/topological_sort.cpp:10]: Dont get what you mean by this

File: Computational_Algorithms/Graphs/topological_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
+#include<bits/stdc++.h>
+using namespace std;
+//B22147 Yash Sehgal
```
Comment (metabiswadeep) [inline Computational_Algorithms/Graphs/topological_sort.cpp:3]: Remove your details from here

Comment (yashsehgal29) [reply Computational_Algorithms/Graphs/topological_sort.cpp:3]: where to mention the details?

Comment (yashsehgal29) [reply Computational_Algorithms/Graphs/topological_sort.cpp:3]: Yash Sehgal
B22147

Comment (metabiswadeep) [reply Computational_Algorithms/Graphs/topological_sort.cpp:3]: in the comments of the PR not the code.

**Comment by metabiswadeep** [comment]
> The code defines a C++ program to perform a topological sort on a directed graph. It uses an adjacency list representation with an unordered_map. The Graph class has methods to add edges, print the graph, and perform a topological sort. The topological sort is implemented using Depth-First Search (DFS). It starts by initializing an array to track visited nodes, then iterates through each node, invoking the DFS function to explore the graph, pushing nodes onto a stack as they are fully explored. Finally, it retrieves nodes from the stack to obtain the topologically sorted order of the graph. The program reads the number of nodes, edges, and edge information from the user and then displays the sorted order.

And add this to the code as comments.

**Comment by yashsehgal29** [comment]
done

**Comment by metabiswadeep** [comment]
Running the code gives

enter the number of nodes and edges : 5 6
Enter the edges : 
1 2
2 3
3 4
4 5
5 1
1 4
5->1,
4->5,
3->4,
2->3,
1->2,4,

The topological  sort : 
1 2 3 4 0

Please recheck your code

**Comment by yashsehgal29** [comment]
the code is implemented as per the data such that the node numbering starts from 0 so try to include 0 as first node

**Comment by yashsehgal29** [comment]
changes in code implemented

**Comment by yashsehgal29** [comment]
removed the unwanted comments

**Pull request merged**